src/prommis/costing/uq/diafiltration_cost_uq.py

```diff
@@ -0,0 +1,1377 @@
+
+Overview of the workflow:
+- Solve a deterministic diafiltration + QGESS cost flowsheet to get an optimal baseline design.
+- Identify a subset of QGESS cost parameters and model them as uncertain.
+- Samples uncertain parameters from specified probability distributions (triangular, uniform, discrete, 
+   or lognormal) using Monte Carlo or Latin Hypercube sampling.
+- Re-solve the flowsheet for each sampled parameter set and record the resulting cost_of_recovery.
+- Produces summary outputs, including:
+    (1) Stage1 membrane bare-erected cost (BEC) versus membrane length, with confidence intervals;
+    (2) The distribution of cost_of_recovery;
+    (3) The distribution of optimized membrane lengths;
+    (4) Summary statistics and selected sensitivity analysis results.
+"""
+
+__author__ = "L. Deng, B. Paul, A. Fritz, A. Garciadiego, A. Ostace, and M. Zamarripa"
+__version__ = "1.0.0"
+
+import math
+import os
+
+import pyomo.environ as pyo
+from pyomo.core.base.param import ScalarParam
+
+from idaes.core import UnitModelBlock, UnitModelCostingBlock
+from idaes.core.util.model_diagnostics import DiagnosticsToolbox, degrees_of_freedom
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+from prommis.nanofiltration.costing.diafiltration_cost_model import (
+    DiafiltrationCostingData,
+)
+from prommis.nanofiltration.diafiltration import (
+    add_product_constraints,
+    build_model,
+    initialize_model,
+    unfix_opt_variables,
+)
+from prommis.uky.costing.costing_dictionaries import load_default_sale_prices
+from prommis.uky.costing.ree_plant_capcost import QGESSCosting
+
+# Make sure ScalarParam has attribute units
+if not hasattr(ScalarParam, "units"):
+    ScalarParam.units = property(lambda self: pyo.units.get_units(self))
+
+
+def get_script_dir():
+    """Return the directory where this script lives."""
+    return os.path.dirname(os.path.abspath(__file__))
+
+
+# 0 Bounds for decision variables
+def decision_variables_bounds(m):
+    m.fs.stage1.length.setlb(0.1)
+    m.fs.stage1.length.setub(10000)
+
+    m.fs.stage2.length.setlb(0.1)
+    m.fs.stage2.length.setub(10000)
+
+    m.fs.stage3.length.setlb(0.1)
+    m.fs.stage3.length.setub(10000)
+
+
+# Sieving coefficient pairs for technology comparison:
+# Case A = (1.3, 0.5), Case B = (1.5, 0.8)
+def set_sieving_coefficients(m, li_sc, co_sc):
+    """
+    Override the Li/Co sieving coefficients on an already constructed diafiltration model.
+
+    Note:
+    The function prommis.nanofiltration.diafiltration.build_model() assigns default
+    sieving coefficient values within diafiltration.py. These defaults are intentionally
+    overridden here, to avoid modifying the underlying flowsheet implementation.
+    """
+    sc = m.fs.sieving_coefficient
+    sc["Li"].unfix()
+    sc["Co"].unfix()
+    sc["Li"].set_value(li_sc)
+    sc["Co"].set_value(co_sc)
+    sc["Li"].fix()
+    sc["Co"].fix()
```

Comment:
```suggestion
    sc["Li"].fix(li_sc)
    sc["Co"].fix(co_sc)
```
Can we just use .fix and eliminate a few unnecessary lines?